test/boost/database_test.cc

```diff
@@ -73,16 +73,20 @@ static future<> apply_mutation(sharded<replica::database>& sharded_db, table_id
     });
 }
 
+future<> do_with_cql_env_and_compaction_groups_cgs(unsigned cgs, std::function<void(cql_test_env&)> func, cql_test_config cfg = {}, thread_attributes thread_attr = {}) {
+    // clean the dir before running
+    if (cfg.db_config->data_file_directories.is_set()) {
+        co_await recursive_remove_directory(fs::path(cfg.db_config->data_file_directories()[0]));
+        co_await recursive_touch_directory(cfg.db_config->data_file_directories()[0]);
+    }
+    cfg.db_config->x_log2_compaction_groups(cgs);
+    co_await do_with_cql_env_thread(func, cfg, thread_attr);
+}
+
 future<> do_with_cql_env_and_compaction_groups(std::function<void(cql_test_env&)> func, cql_test_config cfg = {}, thread_attributes thread_attr = {}) {
     std::vector<unsigned> x_log2_compaction_group_values = { 0 /* 1 CG */, 1 /* 2 CGs */ };
     for (auto x_log2_compaction_groups : x_log2_compaction_group_values) {
-        // clean the dir before running
-        if (cfg.db_config->data_file_directories.is_set()) {
-            co_await recursive_remove_directory(fs::path(cfg.db_config->data_file_directories()[0]));
-            co_await recursive_touch_directory(cfg.db_config->data_file_directories()[0]);
-        }
-        cfg.db_config->x_log2_compaction_groups(x_log2_compaction_groups);
-        co_await do_with_cql_env_thread(func, cfg, thread_attr);
+        co_await do_with_cql_env_and_compaction_groups_cgs(x_log2_compaction_groups, func, cfg, thread_attr);
     }
```

Comment:
Alternatively, you could have used coroutine::parallel_for_each over each cg value, but the way you split it provides even better granularity, as we can see easily when the unit test finished, and which one failed.

Reply:
It won't be faster, since the test is CPU bound.

Reply:
Oops, you're right :-)